extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -188,6 +202,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.lifecycleReporter_.sendPing('layoutCallback');
```

Comment:
No, we need to send them at constructor, `onLayoutMeasure`, and `layoutCallback` in 3p.  The real point of this exercise is to get comparable baselines between the two implementations.  Since those are key methods for the a4a impl, we need to see how often those methods are firing for 3p as well, so that we can disentangle possible implementation issues in A4A from issues elsewhere in the ads pipeline.